batch/create-batch-tables.sql

```diff
@@ -59,9 +61,9 @@ CREATE TRIGGER trigger_jobs_insert AFTER INSERT ON jobs
         UPDATE batch SET n_jobs = n_jobs + 1 WHERE id = new.batch_id;
         IF (NEW.state LIKE 'Complete' OR NEW.state LIKE 'Cancelled') THEN
             UPDATE batch SET n_completed = n_completed + 1 WHERE id = NEW.batch_id;
-            IF (NEW.state LIKE 'Complete' AND NEW.exit_code > 0) THEN
+            IF (NEW.state LIKE 'Complete' AND (NEW.input_exit_code > 0 OR NEW.main_exit_code > 0 OR NEW.output_exit_code > 0)) THEN
 	        UPDATE batch SET n_failed = n_failed + 1 WHERE id = NEW.batch_id;
-            ELSEIF (NEW.state LIKE 'Complete' AND NEW.exit_code = 0) THEN
+            ELSEIF (NEW.state LIKE 'Complete') THEN
```

Comment:
I'm not sure if this is a complete check. Worried about the case where all exit codes are null. But then I don't think it can be complete?

Reply:
I think this is correct.  You can't be complete without main_exit_code being non-missing.